validator/src/consensus/storage/sqlite.ts

```diff
@@ -0,0 +1,293 @@
+import Sqlite3, { type Database } from "better-sqlite3";
+import type { Address, Hex } from "viem";
+import { z } from "zod";
+import { pointFromHex, scalarFromHex, scalarToHex } from "../../frost/math.js";
+import type {
+	FrostPoint,
+	GroupId,
+	ParticipantId,
+	SignatureId,
+} from "../../frost/types.js";
+import { checkedAddressSchema, hexDataSchema } from "../../types/schemas.js";
+import type { NonceTree, PublicNonceCommitments } from "../signing/nonces.js";
+import type {
+	GroupInfoStorage,
+	KeyGenInfoStorage,
+	Participant,
+	SignatureRequestStorage,
+} from "./types.js";
+
+const dbIntegerSchema = z.int().transform((id) => BigInt(id));
+const dbParticipantSchema = z.object({
+	id: dbIntegerSchema,
+	address: checkedAddressSchema,
+});
+const dbPointSchema = z.hex().transform(pointFromHex);
+const dbScalarSchema = z.hex().transform(scalarFromHex);
+
+interface ZodSchema<Output> {
+	parse(data: unknown): Output;
+}
+
+export class SqliteStorage
+	implements KeyGenInfoStorage, GroupInfoStorage, SignatureRequestStorage
+{
+	#account: Address;
+	#db: Database;
+
+	constructor(account: Address, path: string) {
+		const db = new Sqlite3(path);
+		db.exec(`
+			CREATE TABLE IF NOT EXISTS groups(
+				id TEXT NOT NULL,
+				threshold INTEGER NOT NULL,
+				public_key TEXT,
+				verification_share TEXT,
```

Comment:
Related to the above:

It might make sense to separate the verification_share and signing_share from the group and attach a "participantId" to it. This way it would be clear for which participant this info is.